### PR TITLE
Add option to use doc/make_doc

### LINKED ISSUE #3: Support other ways of compiling the documentation
Some packages (e.g. grape, sonata) have a `doc/make_doc` script but not a `makedoc.g` file. This means that this action wouldn't work for these packages, as is.

A quick search reveals the following list of `doc/make_doc` scripts, although I haven't checked whether they all lack a `makedoc.g` file::
```
ace/doc/make_doc
aclib/doc/make_doc
alnuth/doc/make_doc
automgrp/doc/make_doc
autpgrp/doc/make_doc
cohomolo/doc/make_doc
cryst/doc/make_doc
crystcat/doc/make_doc
cubefree/doc/make_doc
design/doc/make_doc
fga/doc/make_doc
format/doc/make_doc
fwtree/doc/make_doc
grape/doc/make_doc
grpconst/doc/make_doc
itc/doc/make_doc
liepring/doc/make_doc
matgrp/doc/make_doc
modisom/doc/make_doc
nilmat/doc/make_doc
radiroot/doc/make_doc
rds/doc/make_doc
sglppow/doc/make_doc
sonata/doc/make_doc
symbCompCC/doc/make_doc
transgrp/doc/make_doc
unipot/doc/make_doc
```

### PR DESCRIPTION
**Pull request by ssiccha**
@wilfwilson do you have a package in mind with which we could test this?

Fixes #3.

### CODE REVIEW

**Comment by wilfwilson** [comment]
I would add the doc-compilation action to this PR https://github.com/gap-packages/sonata/pull/21 and we'd be able to test it there (the result of which we'll be able to see here https://github.com/wilfwilson/sonata/pull/1 until Sonata turns on GitHub Actions)

**Comment by wilfwilson** [comment]
See https://github.com/wilfwilson/sonata/pull/2 for my attempt at getting this to work with sonata package.

**Comment by wilfwilson** [comment]
This seems to be a bit tricky to get to work.

For starters I think the `if: ${{!inputs.use-makedoc-shell-script}}` syntax might be wrong (although I don't know what's the right way of doing it, if it is wrong).

Also the `make_doc` script *possibly* requires the GAP manuals to already be compiled... I am trying to work out if that is the case.

**Comment by ssiccha** [comment]
Ok, can you ping me, when you got the call of make_doc in sonata working? I'll have a look at the `if` syntax in the meanwhile.

File: action.yml [hunk 1]
```diff
@@ -21,8 +25,19 @@ runs:
             sudo apt-get install "${packages[@]}"
           fi
         shell: bash
-      - name: "Compile documentation"
+      - name: "Compile documentation via makedoc.g"
+        if: ${{!inputs.use-makedoc-shell-script}}
```
Comment (wilfwilson) [inline action.yml:29]: I don't think doing `if:` is allowed within a `runs.steps` item of a GitHub workflow (like this one) that's called from other workflow (as e.g. sonata will be).

But I can't get good information on this so far. I'm going to make an alternative PR that works, but which we might want to improve upon.

**Comment by wilfwilson** [comment]
> Ok, can you ping me, when you got the call of make_doc in sonata working? I'll have a look at the `if` syntax in the meantime.

Sorry @ssiccha, I didn't see this until now. I've just made PR #4; I got the compilation to work fine in sonata (GAP's manual doesn't need to be compiled; I was just in the wrong directory; `make_doc` has to be called from within the package's `doc` directory, rather than from the package's root directory (as for `makedoc.g`). See https://github.com/wilfwilson/sonata/pull/2 for a working version with PR #4.

**Comment by wilfwilson** [comment]
I've merged #5 to get a working version into the main branch, but that doesn't mean that this PR is unnecessary.

My PR doesn't add the ssh-on-error thing.

My PR also doesn't actually add an option, instead it just looks for a `makedoc.g` file and runs that if it exists, else it looks for `doc/make_doc` and runs that if it exists, else it fails. So perhaps we still want it to be an option.

**Comment by ssiccha** [comment]
I think trying first `makedoc.g` and `doc/make_doc` as in #5 is more convenient than having an extra option for that.

**Pull request closed without merging**